pkg/management/component.go

```diff
@@ -506,6 +508,32 @@ func (c *component) GetGroupsOfUser(ctx context.Context, realmName, userID strin
 	return groupsRep, nil
 }
 
+func (c *component) GetAvailableTrustIDGroups(ctx context.Context, realmName string) ([]string, error) {
+	var res []string
+	for key := range c.authorizedTrustIDGroups {
+		res = append(res, key)
+	}
+	return res, nil
+}
+
+func (c *component) GetTrustIDGroups(ctx context.Context, realmName, userID string) ([]string, error) {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+	var currentUser, err = c.keycloakClient.GetUser(accessToken, realmName, userID)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return nil, err
+	}
+	var groups = make([]string, 0)
+	for _, grp := range currentUser.GetAttribute(constants.AttrbTrustIDGroups) {
+		if strings.HasPrefix(grp, "/") {
+			grp = grp[1:]
+		}
+		groups = append(groups, grp)
```

Comment:
can it be that one user group is not in the list of c.authorizedTrustIDGroups ?

Reply:
We spoke of this with @harture and decided to ignore this case. Back office will be able to display a warning

Reply:
if BO will do this, then maybe we can be a bit ignorant here :)